### DISCUSSION TITLE
Recommended Web api or MQTT

### DISCUSSION THREAD

**Discussion by felersgit** [0x2142/frigate-notify#65] [General]
I have used both ways and trying to figure out what is the best practice or recommended one to use?  MQTT or the web api?  I am fairly new to this and tried both ways and they work but don’t know what is the best practice for use with pushover.

Thanks for any hints or tips on this.

**Comment by 0x2142** [comment]
Hello there! Thanks for using this app! 

Mostly it's up to personal choice, and whether or not you want/need to run & maintain an MQTT broker. Originally when I built this I didn't have a need for MQTT, so I used the direct web API to query for events. These days, I'm running more in my home lab that utilizes MQTT - so I have swapped over to that.

The MQTT method will be a bit more effective for alerting though. Where the web API method will only query for new events at a specified interval, MQTT events are more real-time. So as Frigate detects events, they're immediately sent to this app for you to be notified. In addition, Frigate can send multiple MQTT messages based on event changes - and therefore we can send new notifications whenever certain things change within the event (like the detected object moving into a new zone, for example). 

So, if you're open to either method - then I would probably suggest MQTT. But both ways will work!

**Comment by felersgit** [comment]
Thank you for taking the time to explain the differences for me!  It really does help me decide on which method I am going to stick to.  I already have a MQTT broker so I will just keep using it then.

Also thanks for making this App, I have now been able to get rid of my home assistant VM which I only used for the notifications!
